pkg/vsphere/compute/rp.go

```diff
@@ -88,3 +88,51 @@ func (rp *ResourcePool) GetCluster(ctx context.Context) (*object.ComputeResource
 
 	return object.NewComputeResource(rp.Client.Client, mrp.Owner), nil
 }
+
+func (rp *ResourcePool) GetDatacenter(ctx context.Context) (*object.Datacenter, error) {
+	dcRef, err := rp.getLowestAncestor(ctx, "Datacenter")
+
+	if err != nil || dcRef == nil {
+		log.Errorf("Unable to get datacenter ancestor of rp %s: %s", rp.Name(), err)
+		return nil, err
+	}
+
+	return object.NewDatacenter(rp.Client.Client, *dcRef), nil
+}
+
+func (rp *ResourcePool) getAncestors(ctx context.Context, inType string) ([]types.ManagedObjectReference, error) {
+	client := rp.Session.Vim25()
+
+	ancestors, err := mo.Ancestors(ctx, client, client.ServiceContent.PropertyCollector, rp.Reference())
+
```

Comment:
Unneeded whitespace.

Reply:
Not sure on which line.

Reply:
Line 107. I think it's a good practice to do this
```
err := funcErr()
if err != nil {
```

instead of
```
err := funcErr()

if err != nil {
```